kubernetes/lib/sar.rego

```diff
@@ -0,0 +1,36 @@
+package kubernetes.lib
+
+import data.kubernetes.lib
+
+sar_raw(resourceAttributes, user, groups) = output {
```

Comment:
Not sure what your use case is exactly.  Typically with OPA you put the information the policy is making a decision about into the `input` document.   Then instead of a function you'll write something like...
```
sar_raw = output {
    body := {... 
               "resourceAttributes": input.resourceAttributes, 
               "user": input.user, 
               "groups": input.groups ... }
    req := ...
    output := http.send(req)
}
```

Reply:
See https://github.com/open-policy-agent/opa/issues/1130 for the use case.

I'm trying to make it easy to write some opa policy that applies just to k8s that can be overridden easily from k8s roles.

Reply:
I'm expecting the users policy to call the function repeatedly in different parts of the policy and hopefully not actually need to issue the http.send most of the time. Does switching it to the format you have above still work for that? I guess I'm struggling a little bit between what a function is and what a rule is.

Reply:
Then a function makes sense here.

A function is something I only use when I am forced to.  Meaning that there's some logic shared between several say deny statements and that logic takes as argument something with an infinite domain, like strings or lists.  

Otherwise I use a rule (set/dictionary/boolean).  A rule is nice because I can decide to include any number of the arguments that I want, and OPA will spit out all the answers for the missing arguments.